internal/signalfx-agent/pkg/monitors/prometheusexporter/prometheus.go

```diff
@@ -133,30 +134,37 @@ func (m *Monitor) Configure(conf *Config) error {
 
 	var ctx context.Context
 	ctx, m.cancel = context.WithCancel(context.Background())
+	applyDimensions := func(attrs pcommon.Map) {
+		attrs.PutStr("system.type", m.monitorName)
+		for k, v := range conf.ExtraDimensions {
+			attrs.PutStr(k, v)
+		}
+	}
 	utils.RunOnInterval(ctx, func() {
-		dps, err := fetchPrometheusMetrics(fetch)
+		metrics, err := fetchPrometheusMetrics(fetch, applyDimensions)
 		if err != nil {
 			m.logger.WithError(err).Error("Could not get prometheus metrics")
 			return
 		}
 
-		m.Output.SendDatapoints(dps...)
+		m.Output.SendMetrics(metrics)
 	}, time.Duration(conf.IntervalSeconds)*time.Second)
 
 	return nil
 }
 
-func fetchPrometheusMetrics(fetch fetcher) ([]*datapoint.Datapoint, error) {
+func fetchPrometheusMetrics(fetch fetcher, applyDimensions func(attrs pcommon.Map)) (pmetric.Metrics, error) {
+	metrics := pmetric.NewMetrics()
 	metricFamilies, err := doFetch(fetch)
 	if err != nil {
-		return nil, err
+		return metrics, err
 	}
+	sm := metrics.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty()
```

Comment:
We probably shouldn't do this for empty `metricFamilies` Let's return early if that's the case?

Reply:
Done.